lib/util.js

```diff
@@ -317,7 +317,7 @@ function inherits(ctor, superCtor) {
 
   if (superCtor.prototype === undefined) {
     throw new ERR_INVALID_ARG_TYPE('superCtor.prototype',
-                                   'Function', superCtor.prototype);
+                                   'Object', superCtor.prototype);
```

Comment:
@BridgeAR thanks for the review! i'm excited to land a commit 😄 

I don't agree with the suggested change because the value of `superCtor.prototype` must be an object. Sure, a function is an object so that works, but its not the function-ness of this value that is important, it is the object-ness (the intention is that this value's properties become available in the prototype chain, that's an object-y aspect of this value). I could apply your reasoning here and say that the suggested change should include `Array`, `RegExp`, etc, since they are also all objects and are equally meaningful here as `Function`, but I think you see that it wouldn't make the error any more meaningful to include those.

Also, I'm not sure which function you mean is deprecated. From the public docs, I don't see any deprecation for `util.inherits()`. Do you mean that `ERR_INVALID_ARG_TYPE()` is deprecated? If so, what is the more favorable alternative?

Thanks for your help!